src/sentry/servermode.py

```diff
@@ -0,0 +1,98 @@
+from __future__ import annotations
+
+import abc
+import functools
+import itertools
+from enum import Enum
+from typing import Any, Callable, Iterable
+
+from django.conf import settings
+
+
+class ServerComponentMode(Enum):
+    """Defines the component mode of the application to be acting as."""
+
+    MONOLITH = "MONOLITH"
+    CONTROL = "CONTROL"
+    CUSTOMER = "CUSTOMER"
+    FRONTEND = "FRONTEND"
```

Comment:
What is the FRONTEND mode for?

Reply:
@markstory it's for running Sentry as a static file webserver.

Reply:
Yeah, it's kind of a placeholder as I don't know if it will actually be needed on any of the individual request handlers mapped in Django. If the front-end silo's *only* job is serving static files, then by definition there is no business logic to tag. This is here because I wasn't sure if that assumption would hold. But, if it does, we can delete the enum value.